plugins/3scale-backend/config.d.ts

```diff
@@ -0,0 +1,19 @@
+import { TaskScheduleDefinitionConfig } from '@backstage/backend-tasks';
+
+interface ThreeScaleConfig {
+  baseUrl: string;
+  serviceAccount: string;
+  serviceAccountCredentials: string;
+  systemLabel?: string;
+  ownerLabel?: string;
+  addLabels?: boolean;
+  schedule?: TaskScheduleDefinitionConfig;
+}
+
+export interface Config {
+  catalog?: {
+    providers?: {
+      threeScaleApiEntity?: Record<string, ThreeScaleConfig>;
```

Comment:
I've played around with this and for some reason `Record<...>` is not generating a validating JSON schema. Current code generates following definition:


```
$schema: https://backstage.io/schema/config-v1
title: Application Configuration Schema
type: object
definitions:
  "Record<string,ThreeScaleConfig>":
    type: object
...
  catalog:
    type: object
    description: Configuration options for the catalog plugin.
    properties:
      providers:
        type: object
        properties:
          threeScaleApiEntity:
            $ref: "#/definitions/Record<string,ThreeScaleConfig>"
```

Compared to `{ [key: string]: ThreeScaleConfig}`:

```
$schema: https://backstage.io/schema/config-v1
title: Application Configuration Schema
type: object
definitions:
  ThreeScaleConfig:
    type: object
    required:
      - baseUrl
      - serviceAccount
      - serviceAccountCredentials
    properties:
      baseUrl:
        type: string
      serviceAccount:
        type: string
      serviceAccountCredentials:
        type: string
      systemLabel:
        type: string
      ownerLabel:
        type: string
      addLabels:
        type: boolean
      schedule: {}
...
  catalog:
    type: object
    description: Configuration options for the catalog plugin.
    properties:
      providers:
        type: object
        properties:
          threeScaleApiEntity:
            type: object
            additionalProperties:
              $ref: "#/definitions/ThreeScaleConfig"
```

Therefore I would suggest to change this to:

```suggestion
      threeScaleApiEntity?: {[key: string]: ThreeScaleConfig};
```

You can check the generated schema via `npx backstage-cli config:schema`

Reply:
Based on my investigation this tracks back to https://github.com/YousefED/typescript-json-schema/issues/337 bug.

Reply:
Done, This is great I'm learning a lot from your reviews. Thanks